crypto.go

```diff
@@ -738,6 +739,124 @@ func (s sikeScheme) setEphemeralKeyPair(skE KEMPrivateKey) {
 	panic("SIKE cannot use a pre-set ephemeral key pair")
 }
 
+///////////
+// Kyber512
+
+type kyber512PublicKey struct {
+	pub *kyber512.PublicKey
+}
+
+type kyber512PrivateKey struct {
+	priv *kyber512.PrivateKey
+	pub  *kyber512.PublicKey
```

Comment:
No need to store the public key along: kyberx.PrivateKey already contains a kyberx.PublicKey. Use the Public() function.